codegen/options.go

```diff
@@ -0,0 +1,54 @@
+package codegen
+
+import (
+	"github.com/rs/zerolog"
+)
+
+// WithNameTransformer overrides how column name will be determined.
+func WithNameTransformer(transformer NameTransformer) TableOptions {
+	return func(t *TableDefinition) {
+		t.nameTransformer = transformer
+	}
+}
+
+// WithSkipFields allows to specify what struct fields should be skipped.
+func WithSkipFields(fields []string) TableOptions {
+	return func(t *TableDefinition) {
+		t.skipFields = fields
+	}
+}
+
+// WithExtraColumns allows passing additional ColumnDefinitions
+func WithExtraColumns(columns ColumnDefinitions) TableOptions {
+	return func(t *TableDefinition) {
+		t.extraColumns = columns
+	}
+}
+
+// WithUnwrapFieldsStructs allows to unwrap specific struct fields (1 level deep only)
+func WithUnwrapFieldsStructs(fields []string) TableOptions {
```

Comment:
I think this slipped through previous review, but shouldn't this be `WithUnwrapStructFields`? 🤔

Reply:
Will fix in the next PR